### ISSUE TITLE
Block website based on Tab

### ISSUE THREAD

**Issue by Naggischnulu** [proginosko/LeechBlockNG#230]
Hi, I'm wondering if it's possible (or if not, then this is a feature request) to block/allow a website based on the tab it's opened in.  I'm using the Firefox multi-account container, so in my case I have multiple tabs of YouTube with different accounts and I want to block only one of them. The problem is that they all use the same URL. One solution could be that LeechBlock can block websites based on the tab name and not just the URL, the tab name can be renamed tab-specific relatively easily with another addon, however I have not found such a solution for the URL.

**Comment by proginosko** [comment]
One possible workaround would be to block by keyword, using the YT account username as the keyword, e.g.:

`youtube.com`
`~myusername`

**Comment by Naggischnulu** [comment]
Thanks for your quick reply, I got it working with a little workaround, I changed the language of YouTube on a tab and used a language specific word as a keyword. The username is sadly not always displayed but your reply gave me the idea for the solution. 
Regardless, the keyword function doesn't seem to stop time (since I use a keyword to allow a site), but only to prevent the site from being blocked after the time is up. This seems a bit counterintuitive, since setting a URL as an exception has also the functionality of stopping the time or rather not showing the time at all.
